### PR TITLE
Fix NXP-30146 audit storage directory

### PR DESCRIPTION
**Pull request by kevinleturc**

### CODE REVIEW

**Review by efge — APPROVED** [review]
Ok for a quick fix avoid the NPEs in test

File: modules/platform/nuxeo-audit-storage-directory/src/main/java/org/nuxeo/audit/storage/stream/StreamAuditStorageWriter.java [hunk 1]
```diff
@@ -113,11 +113,15 @@ protected void writeJsonEntriesToAudit(ComputationContext context) {
             if (jsonEntries.isEmpty()) {
                 return;
             }
-            log.debug("Writing {} log entries to the directory audit storage {}.", jsonEntries.size(),
-                    DirectoryAuditStorage.NAME);
             NXAuditEventsService audit = (NXAuditEventsService) Framework.getRuntime()
                                                                          .getComponent(NXAuditEventsService.NAME);
             DirectoryAuditStorage storage = (DirectoryAuditStorage) audit.getAuditStorage(DirectoryAuditStorage.NAME);
+            if (storage == null) {
```
Comment (bdelbosc) [inline modules/platform/nuxeo-audit-storage-directory/src/main/java/org/nuxeo/audit/storage/stream/StreamAuditStorageWriter.java:119]: my understanding is that the NPE was on the above line in auidt.getAuditStorage ?
```java.lang.NullPointerException: null
	at org.nuxeo.ecm.platform.audit.service.NXAuditEventsService.getAuditStorage(NXAuditEventsService.java:214) ~[nuxeo-platform-audit-core-PR-4633-11.5-SNAPSHOT.jar:?]
	at org.nuxeo.audit.storage.stream.StreamAuditStorageWriter$AuditStorageLogWriterComputation.writeJsonEntriesToAudit(StreamAuditStorageWriter.java:122) ~[nuxeo-audit-storage-directory-PR-4633-11.5-SNAPSHOT.jar:?]
```

Comment (kevinleturc) [reply modules/platform/nuxeo-audit-storage-directory/src/main/java/org/nuxeo/audit/storage/stream/StreamAuditStorageWriter.java:119]: Indeed, I moved the null by initializing the map in `getAuditStorage` in the `Component#start` method, so we need to check if the storage is initialized/present there now.

Comment (bdelbosc) [reply modules/platform/nuxeo-audit-storage-directory/src/main/java/org/nuxeo/audit/storage/stream/StreamAuditStorageWriter.java:119]: perfect then

**Review by bdelbosc — APPROVED** [review]

**Pull request merged**